pqarrow/arrowutils/schema.go

```diff
@@ -0,0 +1,94 @@
+package arrowutils
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/apache/arrow/go/v15/arrow"
+	"github.com/apache/arrow/go/v15/arrow/array"
+)
+
+// EnsureSameSchema ensures that all the records have the same schema. In cases
+// where the schema is not equal, virtual null columns are inserted in the
```

Comment:
Modifying the schema of an arrow record isn't supposed to be supported, and I think may even be removed in future releases of Arrow. 

Can we instead create new records with the unified schema instead?

Reply:
Isn't this what this code is doing? 
We merged all columns both in the schemas and the arrays.

Reply:
Hmm okay I suppose it is, but it's modifying the slice records. Could you make it return a new slice instead?

Reply:
Let me know if this is now closer to what you imagined.

Reply:
Great thanks!